tests/sm_int/sm_int.c

```diff
@@ -1517,4 +1592,57 @@ TEST_FUNCTION(STATE_and_API)
     }
 }
 
+typedef struct WAIT_FOR_CANCEL_TAG
+{
+    SM_HANDLE sm;
+    volatile_atomic int32_t cancel_wait;
+} WAIT_FOR_CANCEL;
+
+static int waitForCancel(void* context)
+{
+    WAIT_FOR_CANCEL* cancel_wait_context = context;
+    ASSERT_IS_NOT_NULL(cancel_wait_context);
+    int32_t current_value = interlocked_add(&cancel_wait_context->cancel_wait, 0);
+    sm_exec_begin(cancel_wait_context->sm);
+    if (wait_on_address(&cancel_wait_context->cancel_wait, current_value, UINT32_MAX) != WAIT_ON_ADDRESS_OK)

```

Comment:
>wait_on_address [](http://example.com/codeflow?start=8&length=15)

This can have spurious wakes, either should be `InterlockedHL_WaitForValue` or just do the loop here (check the value with `interlocked_add` again in a loop).